src/main/scala/thurloe/dataaccess/HttpSamDAO.scala

```diff
@@ -0,0 +1,77 @@
+package thurloe.dataaccess
+
+import com.typesafe.scalalogging.LazyLogging
+import okhttp3.Protocol
+import org.broadinstitute.dsde.workbench.client.sam
+import org.broadinstitute.dsde.workbench.client.sam.api.UsersApi
+import org.broadinstitute.dsde.workbench.client.sam.{ApiCallback, ApiClient, ApiException}
+import org.broadinstitute.dsde.workbench.google.GoogleCredentialModes.GoogleCredentialMode
+import org.broadinstitute.dsde.workbench.google.GoogleUtilities.RetryPredicates.when5xx
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.util.{FutureSupport, Retry}
+
+import java.util
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Future, Promise}
+import scala.jdk.CollectionConverters._
+import scala.jdk.DurationConverters._
+
+class HttpSamDAO(baseSamServiceURL: String, credentials: GoogleCredentialMode, timeout: FiniteDuration)
+    extends SamDAO
+    with Retry
+    with LazyLogging
+    with FutureSupport {
+
+  private val samServiceURL = baseSamServiceURL
+
+  private val okHttpClient = new ApiClient().getHttpClient
+
+  val scopes =
+    List("https://www.googleapis.com/auth/userinfo.email", "https://www.googleapis.com/auth/userinfo.profile")
```

Comment:
I grabbed these scopes from a different call to sam, they are necessary for the `credentials.toGoogleCredential`. Maybe we just pass in an empty list since we are going to do our own auth method?